### PR TITLE
Feat/private registry tracking

### PR DESCRIPTION
**Pull request by varun257**
Add tracking to private registry

MANAGER-4327

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/private-registry/create/create.controller.js [hunk 1]
```diff
@@ -7,10 +7,12 @@ export default class {
     this.privateRegistryService = pciPrivateRegistryService;
     this.loading = false;
     this.registry = {};
+    this.selectedPlan = null;
```
Comment (radireddy) [inline packages/manager/modules/pci/src/projects/project/private-registry/create/create.controller.js]: can you move this into $onInit() and also above two lines of initialization

File: packages/manager/modules/pci/src/projects/project/private-registry/create/create.controller.js [hunk 2]
```diff
@@ -7,10 +7,12 @@ export default class {
...
   }
 
   create() {
     this.loading = true;
+    this.trackClick(`PRIVATEREGISTRY_CREATE_PLAN_CREATE_${this.selectedPlan}`);
```
Comment (radireddy) [inline packages/manager/modules/pci/src/projects/project/private-registry/create/create.controller.js]: I see this code in multiple places. Can you move this line to a separate method (call it trackPrivateRegistryCreate() etc) and call it whenever required.

File: packages/manager/modules/pci/src/projects/project/private-registry/create/create.html [hunk 3]
```diff
@@ -15,6 +15,7 @@ <h2 data-translate="private_registry_create"></h2>
     </oui-step-form>
     <oui-step-form
         data-header="{{:: 'private_registry_create_name_cluster' | translate }}"
+        data-on-submit="$ctrl.submitName('PCI_PROJECTS_PRIVATEREGISTRY_CREATE_VERSION_NEXT')"
```
Comment (radireddy) [inline packages/manager/modules/pci/src/projects/project/private-registry/create/create.html]: move "PCI_PROJECTS_PRIVATEREGISTRY_CREATE_VERSION_NEXT" to a constant file

File: packages/manager/modules/pci/src/projects/project/private-registry/private-registry.html [hunk 4]
```diff
@@ -87,30 +88,31 @@
...
                 </oui-action-menu-item>
                 <oui-action-menu-item
-                    data-on-click="$ctrl.updateRegistry($row.id, $row.name)"
+                    data-on-click="$ctrl.updateRegistry($row.id, $row.name); $ctrl.trackClick('PCI_PROJECTS_PRIVATEREGISTRY_UPDATE')"
                 >
                     <span data-translate="private_registry_rename"></span>
                 </oui-action-menu-item>
                 <oui-action-menu-item
-                    data-on-click="$ctrl.deleteRegistry($row.id, $row.name)"
+                    data-on-click="$ctrl.deleteRegistry($row.id, $row.name); $ctrl.trackClick('PCI_PROJECTS_PRIVATEREGISTRY_DELETE')"
```
Comment (radireddy) [inline packages/manager/modules/pci/src/projects/project/private-registry/private-registry.html]: do you want to track on delete initiate or on delete page visit? I think this takes to delete dialog page right?

**Comment by JeremyDec** [comment]
Hello @varun257, can you please merge your two commits, as they have the same message ?

File: packages/manager/modules/pci/src/projects/project/private-registry/delete/delete.html [hunk 5]
```diff
@@ -8,8 +8,8 @@
         data-heading="{{:: 'private_registry_delete' | translate }}"
         data-primary-label="{{:: 'private_registry_common_delete' | translate }}"
         data-secondary-label="{{:: 'common_cancel' | translate }}"
-        data-secondary-action="$ctrl.goBack()"
-        data-on-dismiss="$ctrl.goBack()"
+        data-secondary-action="$ctrl.goBack(); $ctrl.trackClick('PCI_PROJECTS_PRIVATEREGISTRY_DELETE_CANCEL')"
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/projects/project/private-registry/delete/delete.html]: Maybe we could change the goBack method to send a tracking too

File: packages/manager/modules/pci/src/projects/project/private-registry/delete/delete.html [hunk 6]
```diff
@@ -8,8 +8,8 @@
...
+        data-on-dismiss="$ctrl.goBack(); $ctrl.trackClick('PCI_PROJECTS_PRIVATEREGISTRY_DELETE_CLOSE')"
```
Comment (JeremyDec) [inline packages/manager/modules/pci/src/projects/project/private-registry/delete/delete.html]: Maybe these two calls should be set into one public method, it should help readability

File: packages/manager/modules/pci/src/projects/project/private-registry/private-registry.html (hunk 4 above)
Comment (JeremyDec) [reply packages/manager/modules/pci/src/projects/project/private-registry/private-registry.html]: Aside of @radireddy remark, @varun257 if you choose to set the tracking on visit page, you will have to change the type of track, from `atInternet.trackClick` call to `atInternet.trackPage` one

Comment (varun257) [reply packages/manager/modules/pci/src/projects/project/private-registry/private-registry.html]: Already have a `trackPage` for this. This is in addition to the page tracking. This is specifically to track click event of the button.

File: packages/manager/modules/pci/src/projects/project/private-registry/create/create.controller.js (hunk 2 above)
Comment (varun257) [reply packages/manager/modules/pci/src/projects/project/private-registry/create/create.controller.js]: Looks cleaner now

File: packages/manager/modules/pci/src/projects/project/private-registry/api-url/api-url.component.js [hunk 7]
```diff
@@ -2,8 +2,11 @@ import controller from './api-url.controller';
 import template from './api-url.html';
 
 const component = {
-  template,
+  bindings: {
+    trackClick: '<',
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/private-registry/api-url/api-url.component.js]:
```suggestion
    onClick: '<',
```
we could pass another callback than tracking

Comment (varun257) [reply packages/manager/modules/pci/src/projects/project/private-registry/api-url/api-url.component.js]: But i think this be confusing. Its more generic and could be anything.

**Comment by ovh-cds** [comment]
CDS Report installation#10138.0 ✘  
* Stage 1
  * Install ✘

**Comment by marie-j** [comment]
@varun257 It seems that something went wrong while updating the base branch

**Comment by antleblanc** [comment]
@marie-j rebase on `master`.

**Pull request merged**